fegrow/mlp.py

```diff
@@ -0,0 +1,175 @@
+"""Functionality for working with ML force fields."""
+
+import atexit
+import os
+import tempfile
+import urllib.request
+from abc import ABC, abstractmethod
+from typing import Literal
+
+from openff.toolkit import Molecule
+from openmmml import MLPotential as OMMMLPotential
+
+__all__ = [
+    "AVAILABLE_ML_FORCE_FIELDS",
+    "ANI2X",
+    "MACEOFF23Small",
+    "MACEOFF23Medium",
+    "MACEOFF23Large",
+    "EGRET1",
+]
+
+# Model accessed 24/05/25
+_EGRET1_MODEL_URL = (
+    "https://github.com/rowansci/egret-public/raw/227d6641e6851"
+    "eb1037d48712462e4ce61c1518f/compiled_models/EGRET_1.model"
+)
+
+AVAILABLE_ML_FORCE_FIELDS = Literal[
+    "ani2x",
+    "mace-off23-small",
+    "mace-off23-medium",
+    "mace-off23-large",
+    "egret-1",
+]
+
+
+def _check_mace_installed() -> None:
+    """Check that mace-torch is installed."""
+    try:
+        import mace  # noqa: F401
+
+    except ImportError:
+        msg = (
+            "Using a MACE force field requires the `mace_torch` package. "
+            "Please install it with `pip install mace-torch`."
+        )
+        raise ImportError(msg)
+
+
+class _MLForceField(ABC):
```

Comment:
Should be ok to skip the privatising `_` as this is a class that other might want to subclass to provide other MLPs in their software, if they use FEgrow as a library